### PR TITLE
Upgrade golang to 1.22.2

### PR DESCRIPTION
**Pull request by akash-gautam**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Update to Go 1.22.2
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by embik** [comment]
Note: You'll probably want to do this for `release/v2.25` and the respective post-submit updates also need this: https://github.com/kubermatic/infra/tree/main/prow/jobs/dashboard.

File: modules/web/package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     ]
   },
   "overrides": {
-    "socks": "2.7.3"
+    "socks": "2.7.4"
```
Comment (embik) [inline modules/web/package.json:60]: Why change this?

Comment (embik) [reply modules/web/package.json:60]: Okay, sig-api-ui needs to be the judge of that then :D

Comment (ahmedwaleedmalik) [reply modules/web/package.json:60]: Please revert this change

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**